inginious/frontend/course_factory.py

```diff
@@ -113,6 +114,12 @@ def _migrate_legacy_courses(self):
                     cleaned_taskset_descriptor["dispenser_data"] = taskset_descriptor.get("dispenser_data", {})
                 taskset_descriptor["tasksetid"] = courseid
                 taskset_descriptor["admins"] = taskset_descriptor.get("admins", []) + taskset_descriptor.get("tutors", [])
+                if "accessible" in taskset_descriptor:
+                    taskset_descriptor["accessible"] = change_access_structure(taskset_descriptor["accessible"])
+                if "registration" in taskset_descriptor:
+                    taskset_descriptor["registration"] = change_access_structure(taskset_descriptor["registration"])
+
+                taskset_descriptor = dict_data_str_to_datetimes(taskset_descriptor)
```

Comment:
Same, if we only need `dict_data_str_to_datetimes` here, you can move it here to ease later deprecation.

Reply:
It is used in two other places to transform the task accessibilities back into datetimes (during task accessibilities change in course and taskset template).

There maybe is a better place to place these functions nonetheless.

Reply:
This occurrence has been removed. However, it is still used in the previously mentioned places.